### DISCUSSION TITLE
Extract `Input` and inherit widgets out of `Chat`

### DISCUSSION THREAD

**Discussion by Cierra-Runis** [flyerhq/flutter_chat_ui#524] [Q&A] [answered]
## Related Links
- https://github.com/flyerhq/flutter_chat_ui/issues/322
- https://github.com/flyerhq/flutter_chat_ui/discussions/515

## My Idea
I am using `ScaffoldMessenger` to show `SnackBar` for notification, but `SnackBar` will block the `Input`, so I decide to use `Input` as `Scaffold`'s bottomNavigationBar.

But `Input` needs Three inherit widgets, so it become a heavy code, and these inherit widgets are not exported, which need import files below src folder. What's more, it's better to merge three inherit widgets to a single widget contains three variable.

By extracting `Input` and alert developer to add inherit widgets could makes it much better.

**Comment by demchenkoalex** [comment]
So input should be self contained right? you should be able to just take it and use it elsewhere?

**Reply by Cierra-Runis** [reply]
Yes, remove it from `Chat` widget, and just make sure it has the same inherited widget as the `Chat` widget.

**Reply by demchenkoalex** [reply]
I can't remove it from the chat widget, can I? if I do this then people will need to write more code to make everything work? it should be an option rather?

**Reply by Cierra-Runis** [reply]
😢I see... You can just export those Inherit widgets, that's enough. It seems I could implement my own `Chat` widget since `ChatList` widget has exported. Anyway, thanks for your answers~

**Reply by demchenkoalex** [reply]
Just gathering feedback for v2 cause I am working on a full re-design and it will be much more modular. Will make sure to think about input modularity as well. Thanks for your feedback!

**Comment by demchenkoalex** [comment, marked as answer]
v2 has new modular design that I believe fixes the issue https://github.com/flyerhq/flutter_chat_ui/issues/646
